docs/server/diagnostics/best-practices.md

```diff
@@ -0,0 +1,233 @@
+
+### Memory Utilization
+
+As a database, we seek to use memory efficiently for improved processing. Organizations should perform a memory capacity confirmation test to establish baseline utilization, and monitoring should be performed to look for deviations from this baseline.
+
+Further, monitoring at the operating system level to ensure that **memory utilization does not exceed 85% of physical memory** helps mitigate allocation exceptions
+
+### Garbage Collection Pauses
+
+Garbage collection monitoring is largely concerned with gen2 memory, where longer-lived objects are allocated. The length of **application pauses for compacting garbage collection** of this generation should be monitored using the Kurrent Grafana Dashboard. Steadily increasing durations may eventually cause a leader election as the database will be unresponsive to heartbeats during compacting garbage collections. Monitor this metric to be below the configured Heartbeat Timeout value (default is 10 seconds, so for most customers, 8 seconds should be appropriate)
+
+![Garbage Collection Pauses](./images/2-garbage-collection-pauses.png)
+
+**To mitigate the impact of compacting garbage collection**, KurrentDB 25.1 and above automatically uses the ServerGC algorithm. If you are using a version of KurrentDB below 25.1, it can be enabled with the following environment variables:
+
+* DOTNET\_gcServer set to 1  
+* DOTNET\_GCHeapHardLimitPercent set to 3C (which is 60 in HEX)
+
+On startup, KurrentDB will log that it is using ServerGC:
+
+```
+[64940, 1,07:15:04.489,INF] EventStore                     GC:                       3 GENERATIONS IsServerGC: True Latency Mode: Interactive
+```
+
+### CPU Utilization
+
+To avoid thrashing, monitor **sustained CPU utilization remains below 80%**. This can be done at the operating system level, or on the Kurrent Grafana Dashboard
+
+![CPU Utilization](./images/3-cpu-utilization.png)
+
+### Disk Utilization
+
+Kurrent recommends that organizations configure separate disk locations for logs, data, and indexes to avoid one impacting the other. Monitoring of these spaces should be at the operating system level. Ensure that **log and data disk utilizations are under 90%**. **Index disk utilization should be under 40%**, as additional disk space is required when performing index merges
+
+### Projection Progress
+
+Organizations should monitor the Kurrent Grafana Dashboard to ensure that the **Projection Progress is close to or at 100%** for each projection. This ensures that projections are not falling behind, and keeping pace with appends to the database
+
+![Projection Progress](./images/4-projection-progress.png)
+
+If your **Projection Progress is decreasing, contact Kurrent Support** for analysis and recommendations to mitigate
+
+NOTE: On large databases, this metric could show as 100% but still in fact be far behind due to the number of significant digits when dividing large numbers
```

Comment:
Meta comment: to me that suggests we should actually be showing the number of events behind that we are